### PR TITLE
feat/login

### LINKED ISSUE #4: [4.1.2] Login to TSAO page
Upon login, the user shall be presented with a set of possible actions. Depending
on the role assigned by the administrator, the page would show specific actions
available for each role.

Image

For administrators,
- Create Capstone Entry
- Query, Modify and delete Capstone Entry
- Create, Modify and delete Accounts

Image

For other users,
- Create Capstone Entry
- Query and Modify Capstone Entry

Image

### PR DESCRIPTION
**Pull request by isabellepakyishan**
### Administrator login
image
image
---
### User login
image
image
---
Closes #4

### CODE REVIEW

File: js/login.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    const requestOptions = {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify(data),
+    };
+
+    try {
+        const response = await fetch('http://localhost:8000/api/auth', requestOptions);
+
+        if (response.status === 202) {
```
Comment (Kuan-Zher) [inline js/login.js:25]: Add else statement for handling of other status codes

File: js/login.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
+async function sendAuthRequest() {
```
Comment (Kuan-Zher) [inline js/login.js:2]: Add comments to everything pls

File: js/index.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
+window.onload = function (){
```
Comment (Kuan-Zher) [inline js/index.js]: Add comments also pls

**Review by Kuan-Zher — CHANGES_REQUESTED** [review]
Need to add more comments

**Comment by jiachenyee** [comment]
image
Looks like it is failing due to the test case expecting a certain title, @Kuan-Zher I think we should just update the test case

**Comment by Kuan-Zher** [comment]
> image Looks like it is failing due to the test case expecting a certain title, @Kuan-Zher I think we should just update the test case

Sounds good. I have updated the test case and it runs fine now.

File: js/login.js (hunk 1 above)
Comment (isabellepakyishan) [reply js/login.js:25]: @Kuan-Zher, I've added additional conditional else-if statements to handle other status codes already

File: js/login.js (hunk 2 above)
Comment (isabellepakyishan) [reply js/login.js:2]: @Kuan-Zher Added code documentation for the functions here too

File: js/index.js (hunk 3 above)
Comment (isabellepakyishan) [reply js/index.js]: @Kuan-Zher Added code documentation for the functions here too also

**Review by jiachenyee — APPROVED** [review]
lgtm!

**Review by Kuan-Zher — APPROVED** [review]

**Pull request merged**